nixos/modules/config/stub-ld.nix

```diff
@@ -0,0 +1,81 @@
+{ config, lib, pkgs, ... }:
+
+let
+  inherit (lib) last splitString getExe mkOption types mdDoc mkIf optionals;
+  inherit (builtins) elem toFile;
+
+  stub-ld = pkgs: let
+    message = ''
+      You seem to be trying to run a dynamically linked executable that
+      wasn't made for NixOS. This is possible, but not without extra steps,
+      because NixOS, by design, does not have a global library path.
+
+      There are several solutions available:
+
+      1. If the program is packaged in nixpkgs, use that version.
+
+      2. Write a nix expression for the program to package it
+         yourself. There are 3 approaches to this:
+
+         a. Build from source. This is ideal.
```

Comment:
As @mkg20001 mentioned, please add a link to the manual. Also perhaps state that this can be rather complex.